src/treelearner/monotone_constraints.hpp

```diff
@@ -387,13 +387,13 @@ class FastLeafConstraints : public BasicLeafConstraints {
                            const std::vector<SplitInfo>& best_split_per_leaf) {
     int depth = tree->leaf_depth(~tree->left_child(node_idx)) - 1;
 
-    std::vector<int> features_of_splits_going_up_from_original_leaf;
-    std::vector<uint32_t> thresholds_of_splits_going_up_from_original_leaf;
-    std::vector<bool> was_original_leaf_right_child_of_split;
+    std::vector<int>* features_of_splits_going_up_from_original_leaf = new std::vector<int>();
+    std::vector<uint32_t>* thresholds_of_splits_going_up_from_original_leaf = new std::vector<uint32_t>();
+    std::vector<bool>* was_original_leaf_right_child_of_split = new std::vector<bool>();
```

Comment:
memory leaking here.  you can use `std::vector<T> a`, and pass `&a` in function call.

Reply:
That's modified.